release/release.go

```diff
@@ -98,6 +143,164 @@ func buildArchive() {
 	}
 }
 
+func buildMSI() error {
+	win, err := platform.IsWindows()
+	check(err, "check platform type")
+	if !win {
+		return nil
+	}
+	log.Printf("building msi installer\n")
+
+	// set up build directory.
+	msiBuildDir := "msi_build"
+	os.RemoveAll(msiBuildDir)
+	os.MkdirAll(msiBuildDir, os.ModePerm)
+	os.Chdir(msiBuildDir)
+	oldCwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	// we'll want to go back to the original directory, just in case.
+	defer os.Chdir(oldCwd)
+
+	// Copy sasldlls. They need to be in this directory for Wix. Linking will
+	// not work as the dlls are on a different file system.
+	for _, name := range saslDLLs{
+		err := copyFile(
+			filepath.Join(saslDLLsPath, name),
+			name,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	// make links to all the staticFiles. They need to be in this
+	// directory for Wix.
+	for _, name := range msiStaticFiles {
+		err := os.Link(
+			filepath.Join(msiStaticFilesPath, name),
+			name,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, name := range msiFiles {
+		err := os.Link(
+			filepath.Join(msiFilesPath, name),
+			name,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, name := range binaries {
+		err := os.Link(
+			filepath.Join(binariesPath, name),
+			name + ".exe",
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	// Wix requires the directories to end with a separator.
+	cwd += string(os.PathSeparator)
+	wixPath := string(os.PathSeparator) + filepath.Join("wixtools", "bin")
+	wixUIExtPath := filepath.Join(wixPath, "WixUIExtension.dll")
+	projectName := "MongoDB Tools"
+	sourceDir := cwd
+	resourceDir := cwd
+	binDir := cwd
+	objDir := filepath.Join(cwd, "objs") + string(os.PathSeparator)
+	arch := "x64"
+
+	version := getVersion()
+
+	if version > "r49.0" {
```

Comment:
since the release upgrade uuid is just for the msi, can we pull it down here right above or below this block?

Reply:
Sure, actually I think I'll move all the paths and files. I'm always torn when doing devops type things of putting all paths and what not at the top of the file, or in the function.